packages/blocks/src/api/parser.js

```diff
@@ -434,7 +434,8 @@ export function createBlockWithFallback( blockNode ) {
 	if ( name && name.indexOf( 'core/social-link-' ) === 0 ) {
 		// Capture `social-link-wordpress` into `{"service":"wordpress"}`
 		attributes.service = name.substring( 17 );
-		name = 'core/social-link';
+		// Display social link service name for mobile platform
+		name = Platform.OS === 'web' ? 'core/social-link' : name;
```

Comment:
Can you help me understand why the deprecated name needs to be kept in mobile?

My worries:
* Even though we should keep this compat shim for years to preserve older content, the idea is still to help move content forward by remapping block names, so that, when a user edits an old post, the Social Link markup is correctly updated.
* The shim is likely to move out of the parser and be placed somewhere more sustainable (e.g. external compat file). This could be provided by JS filters or some other form of hooking. Once we do that, it would be important that the shim _unconditionally_ reassigns `name` to `core/social-link`.

/cc @lukewalczak @hypest

Reply:
Hey @mcsf, that change was basically needed to display specific social icon name and its icon within toolbar on the mobile platform.
Without this change, it would be confusing as each `social-icon` would have the same label and the same icon after saving the post.

without parser change | with parser change
--- | ---
<img width="503" alt="Screenshot 2020-07-30 at 14 44 08" src="https://user-images.githubusercontent.com/22746080/88924339-6b361580-d273-11ea-86df-a3a38941e964.png"> | <img width="503" alt="Screenshot 2020-07-30 at 14 45 01" src="https://user-images.githubusercontent.com/22746080/88924345-6cffd900-d273-11ea-9926-094f8e2592f9.png">

The introduced change was the easiest solution to fulfil the requirement, but I understand your concerns and will double-check if we can achieve the same effect without parsers changes.

Reply:
However, could you please elaborate more on:

> The shim is likely to move out of the parser and be placed somewhere more sustainable (e.g. external compat file). This could be provided by JS filters or some other form of hooking. Once we do that, it would be important that the shim unconditionally reassigns name to core/social-link

Reply:
Thanks for the context.

It's clear then that support for Block Variations needs to be improved so that those interactions can take into account attributes like a variation's title or icon.

For comparison, even on the Web we let the variation show in certain places and let the base type show in others, which is also a problem but less confusing than in the apps:

<img width="784" alt="Captura de ecrã 2020-07-30, às 14 12 11" src="https://user-images.githubusercontent.com/150562/88939335-4eeba600-d27e-11ea-92b3-e94bb03e58b2.png">

See **open questions** about variations: https://github.com/WordPress/gutenberg/issues/16283#issuecomment-593404741

* * *

> However, could you please elaborate more on:
> > The shim is likely to move out of the parser […]

Sure. I'm referring to [this comment by Grzegorz](https://github.com/WordPress/gutenberg/pull/19887/files#r375181162).

* * *

Context:
* Block variations API: #16283, #19243 
* Social Icons implementation: #19887

Reply:
So I assume that the main goal is to revert that change and have equal HTML on both platforms?

Reply:
Yep, correct.

Reply:
I've created a [PR](https://github.com/WordPress/gutenberg/pull/24306) fixing that, please check it :v: